src/simpleNbox.cpp

```diff
@@ -408,89 +478,140 @@ unitval SimpleNbox::getData(const std::string& varName,
+    } else if(varNameParsed == D_Q10_RH) {
         H_ASSERT(date == Core::undefinedIndex(), "Date not allowed for Q10");
         returnval = unitval(q10_rh, U_UNITLESS);
-    } else if( varName == D_LAND_CFLUX ) {
+    } else if( varNameParsed == D_LAND_CFLUX ) {
         H_ASSERT( date == Core::undefinedIndex(), "Date not allowed for atm-land flux" );
         returnval = sum_npp() - sum_rh() - lucEmissions.get( ODEstartdate );
         
-    } else if( varName == D_RF_T_ALBEDO ) {
+    } else if( varNameParsed == D_RF_T_ALBEDO ) {
         H_ASSERT( date != Core::undefinedIndex(), "Date required for albedo forcing" );
         returnval = Ftalbedo.get( date );
 
         // Partitioning parameters.
         // For now, only global values are supported.
         // TODO Biome-specific versions of all of these
-    } else if(varName == D_F_NPPV) {
+    } else if(varNameParsed == D_F_NPPV) {
         H_ASSERT(date == Core::undefinedIndex(), "Date not allowed for vegetation NPP fraction");
         returnval = unitval(f_nppv, U_UNITLESS);
-    } else if(varName == D_F_NPPD) {
+    } else if(varNameParsed == D_F_NPPD) {
         H_ASSERT(date == Core::undefinedIndex(), "Date not allowed for detritus NPP fraction");
         returnval = unitval(f_nppd, U_UNITLESS);
-    } else if(varName == D_F_LITTERD) {
+    } else if(varNameParsed == D_F_LITTERD) {
         H_ASSERT(date == Core::undefinedIndex(), "Date not allowed for litter-detritus fraction");
         returnval = unitval(f_litterd, U_UNITLESS);
-    } else if(varName == D_F_LUCV) {
+    } else if(varNameParsed == D_F_LUCV) {
         H_ASSERT(date == Core::undefinedIndex(), "Date not allowed for LUC vegetation fraction");
         returnval = unitval(f_lucv, U_UNITLESS);
-    } else if(varName == D_F_LUCD) {
+    } else if(varNameParsed == D_F_LUCD) {
         H_ASSERT(date == Core::undefinedIndex(), "Date not allowed for LUC detritus fraction");
         returnval = unitval(f_lucd, U_UNITLESS);
         
-    } else if( varName == D_EARTHC ) {
+    } else if( varNameParsed == D_EARTHC ) {
         if(date == Core::undefinedIndex())
             returnval = earth_c;
         else
             returnval = earth_c_ts.get(date);
-    } else if( varName == D_VEGC ) {
-        if(date == Core::undefinedIndex())
-            returnval = sum_map( veg_c );
-        else
-            returnval = sum_map(veg_c_tv.get(date));
-    } else if( varName == D_DETRITUSC ) {
-        if(date == Core::undefinedIndex())
-            returnval = sum_map( detritus_c );
-        else
-            returnval = sum_map(detritus_c_tv.get(date));
-    } else if( varName == D_SOILC ) {
-        if(date == Core::undefinedIndex()) 
-            returnval = sum_map( soil_c );
-        else
-            returnval = sum_map(soil_c_tv.get(date));
-    } else if( varName == D_FFI_EMISSIONS ) {
+    } else if( varNameParsed == D_VEGC ) {
+        if(biome == SNBOX_DEFAULT_BIOME) {
```

Comment:
So the subtle (arguably ill-advised? Definitely insufficiently documented) design choice here is that `global.veg_c` in the case where there are multiple biomes will _still work_, and is a special case that returns the sum of vegetation C pools across all biomes. For instance, if you have `veg_c["default"]` and `veg_c["arctic"]`, then asking for `VEG_C()` (same as `VEG_C("global")`) will return `veg_c["default"] + veg_c["arctic"]`.

This may or may not be good design, and while I think the ability to do this is useful in some cases, I also definitely see the rationale for dropping it.

Reply:
Will it work that way?  I thought that earlier in that method we check to see if the asked-for biome exists, and we throw an error if it doesn't (I'm going from memory here, but I think that's how it worked).  If so, then we _can't_ ask for `VEG_C("global")` when multi-biome is in effect because you aren't allowed to have a global biome when you have other biomes.

But if I'm mistaken, and we do let "global" through in the multi-biome case, then I'm ok with it as is.

Reply:
Good catch!

Interestingly, what I had was a situation where `VEG_C()` (`"veg_c"`) would work as expected, but `VEG_C("global")` (`"global.veg_c"`) would not, because the `has_biome` assertion was only run if the split character was detected.

However, having that assertion there was redundant because we run it _again_ before every biome-specific parameter retrieval. So I removed it in 58469cb7875c12ce92d832974b6d02069a05ee44 and added a quick test to make sure that both cases work as expected.